### PR TITLE
Confirm sighting end point

### PR DESCRIPTION
**Pull request by francescastocco**
https://trello.com/c/udi2xr0n

### CODE REVIEW

File: WhaleSpotting.UnitTests/Services/SightingsServiceTests.cs [hunk 1]
```diff
@@ -96,5 +96,51 @@ public void CreateSighting_CalledWithInvalidSightingRequestModel_ThrowsAnExcepti
...
+                OrcaType = null,
+                OrcaPod = "",
+                Confirmed = false,
+            };
+
+            await Context.Sightings.AddAsync(sighting);
+            await Context.SaveChangesAsync();
+
+            // Act
+            var result = _underTest.ConfirmSighting(id);
```
Comment (oskarwilliams) [inline WhaleSpotting.UnitTests/Services/SightingsServiceTests.cs]: you should await this

File: WhaleSpotting.UnitTests/Services/SightingsServiceTests.cs [hunk 2]
```diff
@@ -96,5 +96,51 @@ public void CreateSighting_CalledWithInvalidSightingRequestModel_ThrowsAnExcepti
...
+
+            // Assert
+            result.Should().BeOfType<Task<SightingResponseModel>>();
```
Comment (oskarwilliams) [inline WhaleSpotting.UnitTests/Services/SightingsServiceTests.cs]: This shouldn't have the Task, and can we test 1 property of the sighting response model

File: WhaleSpotting.UnitTests/Services/SightingsServiceTests.cs [hunk 3]
```diff
@@ -96,5 +96,51 @@ public void CreateSighting_CalledWithInvalidSightingRequestModel_ThrowsAnExcepti
...
+        }
+
+        [Fact]
+        public void ConfirmSighting_CalledWithInvalidId_ReturnsNullSightingResponseModel()
+        {
+            // Arrange
+            const int id = 1;
+
+            // Act
+            var nullResult = _underTest.ConfirmSighting(id);
```
Comment (oskarwilliams) [inline WhaleSpotting.UnitTests/Services/SightingsServiceTests.cs]: This should be awaited too

File: WhaleSpotting/Controllers/SightingsController.cs [hunk 4]
```diff
@@ -39,5 +39,12 @@ public IActionResult CreateSighting([FromBody] SightingRequestModel sightingRequ
                 return BadRequest(e.Message);
             }
         }
+
+        [HttpPut("/{id}/confirm")]
+        public async Task<ActionResult<SightingResponseModel>> ConfirmSighting([FromRoute] int id)
+        {
+            var sighting = await _sightings.ConfirmSighting(id);
+            return sighting == null ? NotFound() : sighting;
```
Comment (oskarwilliams) [inline WhaleSpotting/Controllers/SightingsController.cs:49]: nice! :)

File: WhaleSpotting/Controllers/SightingsController.cs [hunk 5]
```diff
@@ -39,5 +39,12 @@ public IActionResult CreateSighting([FromBody] SightingRequestModel sightingRequ
                 return BadRequest(e.Message);
             }
         }
+
+        [HttpPut("/{id}/confirm")]
```
Comment (oskarwilliams) [inline WhaleSpotting/Controllers/SightingsController.cs]: can you get rid of the first slash here, and for the create. Basically with the / the url here is just `localhost:5001/{id}/confirm`, but we want it to be `localhost:5001/sightings/{id}/confirm`

File: WhaleSpotting/Services/SightingsService.cs [hunk 6]
```diff
@@ -64,5 +65,22 @@ public SightingResponseModel CreateSighting(SightingRequestModel sightingRequest
 
             return new SightingResponseModel(newSighting);
         }
+
+        public async Task<SightingResponseModel> ConfirmSighting(int id)
+        {
+            var sighting = await _context.Sightings
+                //TODO include user
```
Comment (oskarwilliams) [inline WhaleSpotting/Services/SightingsService.cs]: Do we need to include the user in the future?

File: WhaleSpotting/Services/SightingsService.cs [hunk 7]
```diff
@@ -64,5 +65,22 @@ public SightingResponseModel CreateSighting(SightingRequestModel sightingRequest
...
+                .SingleOrDefaultAsync(s => s.Id == id);
+
+            if (sighting==null)
```
Comment (oskarwilliams) [inline WhaleSpotting/Services/SightingsService.cs]: can we have spacing around the ==

**Review by oskarwilliams — CHANGES_REQUESTED** [review]

File: WhaleSpotting/Services/SightingsService.cs (hunk 6 above)
Comment (LenaESoftwire) [reply WhaleSpotting/Services/SightingsService.cs]: We wanted for every sighting to have the UserId for the user who posted that sighting.
And for our profile page to show the sightings of that user.
Currently we are just passing the  UserId when creating sightings. Do you think it is enough? Or should we connect them?

Comment (oskarwilliams) [reply WhaleSpotting/Services/SightingsService.cs]: We won't need to return the actual user for just a confirmation of sighting is my thought. For the full return we may want to, but here I would suggest not

**Review by oskarwilliams — CHANGES_REQUESTED** [review]

**Review by oskarwilliams — APPROVED** [review]

**Pull request merged**